core/domain/stats_jobs_one_off.py

```diff
@@ -127,3 +143,2369 @@ def reduce(key, stringified_values):
+                item_id, strict=False)
+            if migrated_answer_record:
+                for exp_version in migrated_answer_record.exploration_versions:
+                    all_models = stats_models.StateAnswersModel.get_all_models(
+                        migrated_answer_record.exploration_id, exp_version,
+                        migrated_answer_record.state_name)
+                    if not all_models:
+                        continue
+                    # The entire state of records need to be deleted since there
+                    # isn't a reliable way to go into submitted answers and
+                    # remove specific failures. The entire state will be
+                    # re-migrated the next time the AnswerMigrationJob runs.
+                    for state_answer_model in all_models:
+                        submitted_answers = [
+                            stats_domain.SubmittedAnswer.from_dict(
+                                submitted_answer_dict)
+                            for submitted_answer_dict
+                            in state_answer_model.submitted_answer_list]
+                        exp_id = state_answer_model.exploration_id
+                        state_name = state_answer_model.state_name
+                        for submited_answer in submitted_answers:
+                            rule_str = submited_answer.rule_spec_str
+                            old_item_id = u'%s.%s.%s.%s' % (
+                                exp_id, state_name, 'submit', rule_str)
+                            separate_migrated_answer_record = (
+                                stats_models.MigratedAnswerModel.get(
+                                    old_item_id, strict=False))
+                            if separate_migrated_answer_record:
+                                separate_migrated_answer_record.delete()
+                        state_answer_model.delete()
+                yield (
+                    'Deleting \'%s\' since its new model is inconsistent with '
+                    'its old model' % item_id.decode('utf-8'))
+
+
+class ClearMigratedAnswersJob(jobs.BaseMapReduceJobManager):
+    """This job deletes all answers stored in the
+    stats_models.StateAnswersModel and all book-keeping information stored in
+    stats_models.MigratedAnswerModel.
+    """
+    _DELETE_KEY = 'Migrated Answer Cleared'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [
+            stats_models.StateAnswersModel, stats_models.MigratedAnswerModel]
+
+    @staticmethod
+    def map(item):
+        item.delete()
+        yield (ClearMigratedAnswersJob._DELETE_KEY, 'Deleted answer.')
+
+    @staticmethod
+    def reduce(key, stringified_values):
+        yield 'Deleted %d answer buckets' % len(stringified_values)
+
+
+class PurgeInconsistentAnswersJob(jobs.BaseMapReduceJobManager):
+    """This job deletes all answers stored in
```

Comment:
Resolving an answer does just get rid of it. No stats come to mind for answers which weren't either computed on the fly (and still will be after this migration) or were pre-aggregated (and will be computed on the fly after this migration).